Main/OOP18/drink.cpp

```diff
@@ -0,0 +1,78 @@
+#include "drink.h"
+#include <iostream>
+
+// Base Drink Class
+Drink::Drink(DrinkType t, SizeType s) : type(t), size(s), cost(0) {}
+
+unsigned int Drink::getCost() const {
+    return cost;
+}
+
+// Coffee Class
+Coffee::Coffee(DrinkType t, SizeType s) : Drink(t, s) {
+    switch (type) {
+    case DrinkType::ESPRESSO:
+        cost = 50;
+        size = SizeType::NONE; // Espresso is one size only
+        break;
+    case DrinkType::AMERICANO:
+        cost = (size == SizeType::M) ? 40 : (size == SizeType::L) ? 60 : 80;
+        break;
+    case DrinkType::LATTE:
+        cost = (size == SizeType::M) ? 70 : (size == SizeType::L) ? 90 : 110;
+        break;
+    case DrinkType::CAPPUCCINO:
+        cost = (size == SizeType::M) ? 80 : 100; // Cappuccino has M and L sizes
+        break;
+    default:
+        cost = 0;
+        break;
+    }
+}
+
+void Coffee::prepare() const {
+    switch (type) {
+    case DrinkType::ESPRESSO:
```

Comment:
coffee alreader has CoffeeType enum, so better use that one as Coffee shouldn't know anything about TEA, but it does as DrinkType also has info about Tea types